src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -121,19 +93,13 @@ export function ThemeProvider({
   );
 }
 
-function isInverseColorScheme(
-  colorScheme?: InversableColorScheme,
-): colorScheme is Inverse {
-  return colorScheme === 'inverse';
-}
-
 function getColorScheme(
-  colorScheme: InversableColorScheme | undefined,
-  parentColorScheme: OriginalColorScheme | undefined,
-) {
+  colorScheme: ThemeConfig['colorScheme'],
+  parentColorScheme?: Theme['colorScheme'],
+): Theme['colorScheme'] {
   if (colorScheme == null) {
     return parentColorScheme || 'light';
-  } else if (isInverseColorScheme(colorScheme)) {
+  } else if (colorScheme === 'inverse') {
```

Comment:
✨  Nice and simple, I'm typically a fan of typeguards but this makes more sense in this case

Reply:
This is still a typeguard, it's just one that TS can work out automatically instead of us having  to be explicit.

Inside this if block the type of `colorScheme` is still narrowed down `'light'| 'dark' | 'inverse'` to `'light' | 'dark'`  (`undefined` was already narrowed away thanks to the prior if block)